functions/Get-DbaServerInstallDate.ps1

```diff
@@ -0,0 +1,181 @@
+		[Alias("ServerInstance", "SqlInstance", "ComputerName")]
+		[DbaInstanceParameter[]]$SqlServer,
+		[PsCredential]$SqlCredential,
+		[PsCredential]$WindowsCredential,
+		[parameter(ParameterSetName = "Sql")]
+		[Switch]$SqlOnly,
+		[parameter(ParameterSetName = "Windows")]
+		[Switch]$WindowsOnly,
+		[Alias("Credential")]
+		[switch]$Silent
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlServer)
+		{
+			if ($instance.Gettype().FullName -eq [System.Management.Automation.PSCustomObject] )
+			{
+				$servername = $instance.SqlInstance
+			}
+			elseif ($instance.Gettype().FullName -eq [Microsoft.SqlServer.Management.Smo.Server])
+			{
+				$servername = $instance.NetName
+			}
+			else
+			{
+				$servername = $instance
+			}
+						
+			if ($WindowsOnly -ne $true)
+			{ 				
+				try 
+				{
+					Write-Message -Level Verbose -Message "Connecting to $instance"
+                    if ($SqlCredential)
+                    { 
+						$server = Connect-SqlInstance -SqlInstance $servername -SqlCredential $SqlCredential
+                    } else 
+                    {
+						$server = Connect-SqlInstance -SqlInstance $servername
+                    }					
+				}
+				catch 
+				{
+					Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance					
+				}
+
+				if ( $server.VersionMajor -ge 9 )
+				{ 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" 
+					$sql = "SELECT create_date FROM sys.server_principals WHERE sid = 0x010100000000000512000000"
+					[DbaDateTime]$sqlInstallDate = $server.Query($sql, 'master', $true).create_date
+
+				} else { 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" 
+					$sql = "SELECT schemadate FROM sysservers"
+					$sqlInstallDate = $server.Query($sql, 'master').tables
+					[DbaDateTime]$sqlInstallDate = $server.Query($sql, 'master', $true).create_date
+				}											
+
+			} 
+
+			if ($SqlOnly -ne $true)			
+			{ 
+				Write-Message -Level Verbose -Message "Getting Windows Server Name for: $servername" 
+				$WindowsServerName = (Resolve-DbaNetworkName $servername -Credential $WindowsCredential).ComputerName
+				
+				try
+				{
+					Write-Message -Level Verbose -Message "Getting Windows Install date via CIM for: $WindowsServerName"
+					[DbaDateTime]$windowsInstallDate = (Get-DbaCmObject -ComputerName $WindowsServerName -ClassName win32_OperatingSystem -Credential $WindowsCredential).InstallDate					
+				}
+				catch
+				{
+					try
+					{	
+						Write-Message -Level Verbose -Message "Getting Windows Install date via DCOM for: $WindowsServerName" 
+						$CimOption = New-CimSessionOption -Protocol DCOM
+						$CimSession = New-CimSession -Credential:$WindowsCredential -ComputerName $WindowsServerName -SessionOption $CimOption
+						[DbaDateTime]$windowsInstallDate = ($CimSession | Get-CimInstance -ClassName Win32_OperatingSystem).InstallDate					
```

Comment:
This actually broke the windows install date return when added, my previous code worked for every server and Local host I used it against, however this doesn't return anything. 

```
if ( $IncludeWindows )			
			{ 
				Write-Message -Level Verbose -Message "Getting Windows Server Name for: $servername" 				
				try
				{
					if (Was-Bound $Credential) {
						$WindowsServerName = (Resolve-DbaNetworkName $servername -Credential $Credential).ComputerName
						[DbaDateTime]$windowsInstallDate = (Get-DbaCmObject -ClassName win32_OperatingSystem -ComputerName $WindowServerName -Credential $Credential -Silent).InstallDate
					}
					else {
						$WindowsServerName = (Resolve-DbaNetworkName $servername ).ComputerName
						[DbaDateTime]$windowsInstallDate = (Get-DbaCmObject -ClassName win32_OperatingSystem -ComputerName $WindowServerName -Silent).InstallDate
					} 				
				}
				catch
				{				
					Stop-Function -Message "Failed to connect to: $WindowsServerName" -Continue -Target $instance -InnerErrorRecord $_
				}
			}

```